src/robot/htmldata/template.py

```diff
@@ -14,16 +14,29 @@
 #  limitations under the License.
 
 import os
-from os.path import abspath, dirname, join, normpath
+import pathlib
+import sys
 
+if sys.version_info < (3, 10) and not os.path.exists(__file__):
+    # Try importlib resources backport as prior to python 3.10
+    # importlib.resources.files was not zipapp compatible...
+    try:
+        from importlib_resources import files
+    except ImportError:
+        raise ImportError("robotframework outside of filesystem (zipapp?) requires importlib resources backport on python < 3.10")
```

Comment:
Style issues:
- Too long line. Our convention is max 88 characters.
- The message is somewhat confusing. You could try rephrasing it, but I can update it later as well.

Reply:
I tried to express that this applies to all environments where robotframework is not installed on a file system. zipapps, pyoxidizer etc... I tried to improve, please take over if it is still not good enough.